meson.build

```diff
@@ -77,7 +77,6 @@ if is_gcc
 	devel_args += [
 		'-fmax-errors=5',
 		'-Wno-init-list-lifetime',
-		'-DTOML_EXTERN_TEMPLATES=0'
```

Comment:
Dropping EXTERN_TEMPLATES=0 did not helped :/

Reply:
Yeah, I misunderstood what was happening before. What I didn't realize before, but do now, is that if you're now seeing `TOML_API`-related issues when building the 'devel' projects that you were not seeing before, then `TOML_API` is now being set to something it shouldn't be. They should not be consuming it as a shared/static library, but just directly via the header like before your changes, so `TOML_API` should not be set to anything.

> The issue was there before this PR,

Probably, for end-user code, sure. I don't doubt it. I've only used `TOML_API` on windows with DLL projects; I have not tested GCC in this scenario at all. Neither here nor there, though, because again, it never applied to the 'devel' projects, because `TOML_API` should not be set there. We can figure out what's going on for `TOML_API` on GCC once the test/CI projects are back to building how they were before.

Reply:
Oh, I see what's changed. This addition is causing the issue:

https://github.com/marzer/tomlplusplus/blob/61b10632f388c1c46f2bf475a935caebdc326fde/include/toml%2B%2B/impl/preprocessor.h#L418-L422

You're unconditionally `__attribute__ ((visibility ("default")))` when compiling with GCC without header-only mode, even when not being consumed as a shared library. I think this is the wrong move. But I think I do understand the issue better now. Lemme think on this and I'll come up with something.

Reply:
> You're unconditionally applying `__attribute__ ((visibility ("default")))` when compiling with GCC without header-only mode, even when not being consumed as a shared library.

Yep. I could set `TOML_API` to `__attribute__ ((visibility ("default")))` only when using the shared library, but that wouldn't change the fact that applying visibility attributes to `extern` variables is not possible with GCC. So the warning would still be there, but only when using the shared library. Compared to MSVC/Windows, using visibility attributes with GCC when using static libraries doesn't affect compilation/linking in particular ways

Reply:
Ok, i know what needs to happen, but it's a bit much to explain here. I'll push some changes to the PR branch.

Reply:
Decided I'm just gonna nuke the TOML_EXTERN_TEMPLATES nonsense since it makes this all much harder to reason about. We can fold it into your PR since it's basically a blocker for you.

Reply:
Ok! Feel free to push your changes in this branch :)

Reply:
Yup, just did, I removed the extern stuff altogether, which allowed me to simplify `preprocessor.h` quite a bit. Basically:

- `TOML_SHARED_LIB == 1` now implies/overrides `TOML_HEADER_ONLY == 0`
- I removed `TOML_SHARED_LIB_BUILDING` altogether, since it was serviing the exact same purpose `TOML_IMPLEMENTATION` was already
- `TOML_API` is only ever set to a non-blank value when `TOML_SHARED_LIB == 1`

That should simplify things enough to unblock you :)

Reply:
Thanks! Overriding `TOML_HEADER_ONLY` makes sense. Regarding only setting `TOML_API` when building/consuming the shared library is I think that's also ok. Lastly, removing `TOML_SHARED_LIB_BUILDING` and only using `TOML_IMPLEMENTATION`... It took me a while to realize it, but you're right :D